File: src/main/java/org/sti/api/domain/Vehiculo.java
package org.sti.api.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "vehiculos")
public class Vehiculo {

    @Id
    @Column(length = 20)
    private String matricula;

    @Column(nullable = false, length = 100)
    private String marca;

    @Column(nullable = false, length = 100)
    private String modelo;

    @Column(name = "anio")
    private int anio;

    @Column(length = 50)
    private String color;

    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @OneToMany(mappedBy = "vehiculo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrdenReparacion> ordenes;
}

File: frontend/src/types/vehiculo.ts
export interface Vehiculo {
  matricula: string;
  marca: string;
  modelo: string;
  anio: number;
  color?: string;
  clienteId: number;
  clienteNombre?: string;
  clienteApellidos?: string;
}

export interface VehiculoInDto {
  matricula: string;
  marca: string;
  modelo: string;
  anio: number;
  color?: string;
  clienteId: number;
}
